CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.22)
```

Comment:
As noted before this seems quite high, can we use a lower min version?

Reply:
LLVM requires 3.20, and check_linker_flag was added in 3.18, so maybe 3.20 is a good base version?

Reply:
Changed to 3.20